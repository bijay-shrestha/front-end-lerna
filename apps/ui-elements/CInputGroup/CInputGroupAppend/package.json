{
    "name": "@cogent/c-inputgroup-append",
    "main":"CInputGroupAppend.js",
    "module": "CInputGroupAppend.js",
    "version": "0.0.0"
}