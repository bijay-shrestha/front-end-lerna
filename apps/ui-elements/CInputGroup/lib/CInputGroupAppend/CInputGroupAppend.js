"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _reactBootstrap = require("react-bootstrap");

var CInputGroupAppend = function CInputGroupAppend(_ref) {
  var id = _ref.id,
      classes = _ref.classes,
      append = _ref.append;
  return _react["default"].createElement(_reactBootstrap.InputGroup.Append, {
    id: id,
    className: classes
  }, append && append.map(function (ap, index) {
    return ap;
  }));
};

var _default = (0, _react.memo)(CInputGroupAppend);

exports["default"] = _default;