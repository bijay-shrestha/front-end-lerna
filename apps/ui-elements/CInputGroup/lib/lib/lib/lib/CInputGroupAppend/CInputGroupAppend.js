"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _reactBootstrap = require("react-bootstrap");

function _interopRequireWildcard(obj) {
  if (obj && obj.__esModule) {
    return obj;
  } else {
    var newObj = {};

    if (obj != null) {
      for (var key in obj) {
        if (Object.prototype.hasOwnProperty.call(obj, key)) {
          var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {};

          if (desc.get || desc.set) {
            Object.defineProperty(newObj, key, desc);
          } else {
            newObj[key] = obj[key];
          }
        }
      }
    }

    newObj["default"] = obj;
    return newObj;
  }
}

var CInputGroupAppend = function CInputGroupAppend(_ref) {
  var id = _ref.id,
      classes = _ref.classes,
      append = _ref.append;
  return _react["default"].createElement(_reactBootstrap.InputGroup.Append, {
    id: id,
    className: classes
  }, append && append.map(function (ap, index) {
    return ap;
  }));
};

var _default = (0, _react.memo)(CInputGroupAppend);

exports["default"] = _default;