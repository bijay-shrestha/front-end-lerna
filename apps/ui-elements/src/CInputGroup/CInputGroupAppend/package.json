{
    "name": "@cogent/c-inputgroup-append",
    "main": "lib/CInputGroupAppend.js",
    "main:src": "CInputGroupAppend.js",
    "module": "CInputGroupAppend.js",
    "version": "0.0.0",
    "license": "MIT"
}