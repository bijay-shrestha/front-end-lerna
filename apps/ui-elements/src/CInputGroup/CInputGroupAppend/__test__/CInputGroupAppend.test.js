import CInputGroupAppend from '../CInputGroupAppend';
import { InputGroup,Button } from 'react-bootstrap';
expect.addSnapshotSerializer(enzymeSerializer);
describe('CInputGroupAppend', () => {
    let wrapper;
    beforeEach(() => {
    wrapper=mount(<CInputGroupAppend id="appendinputGroup"
                       classes="hello"
                       append={''}
                  />)
    });
    afterEach(() => {
      wrapper.unmount();
    })
    it('should be defined',() => {
      expect(wrapper).toBeDefined();
    });

    it('should have given props available' , () => {
      const propsForAppend = ['id','classes','append'];
      const appendPropsAv = Object.keys(wrapper.props())
      propsForAppend.map(( propsForApp , i ) => {
        expect(propsForAppend).toContain(appendPropsAv[i])
      });
    });

    it( 'should have atleast one element of input inside append element' , () => {
      const a =[<InputGroup.Text key="addon-1" id="basic-addon3">
                https://example.com/users/
               </InputGroup.Text>,
              <Button key="button2" id="buton2" variant="outline-secondary">Button</Button>
              ]
     wrapper.setProps({append:a})
     expect(wrapper.find('#basic-addon3').length).toBeGreaterThan(1);
    });

    describe('Snapshot testing' , () => {
     it('should match the previous snapshot' , () => {
        expect(wrapper).toMatchSnapshot();
     })
     
    });

})