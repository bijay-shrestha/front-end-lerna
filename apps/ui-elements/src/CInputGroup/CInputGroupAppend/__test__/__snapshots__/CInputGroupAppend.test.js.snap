// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CInputGroupAppend Snapshot testing should match the previous snapshot 1`] = `
<Memo(CInputGroupAppend)
  append=""
  classes="hello"
  id="appendinputGroup"
>
  <InputGroupAppend
    className="hello"
    id="appendinputGroup"
  >
    <div
      className="hello input-group-append"
      id="appendinputGroup"
    />
  </InputGroupAppend>
</Memo(CInputGroupAppend)>
`;
